go/src/koding/kites/kloud/provider/marathon/schema.go

```diff
@@ -0,0 +1,96 @@
+package marathon
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"time"
+
+	"koding/kites/kloud/stack"
+	"koding/kites/kloud/stack/provider"
+
+	marathon "github.com/gambol99/go-marathon"
+)
+
+var schema = &provider.Schema{
+	NewCredential: func() interface{} {
+		return &Credential{}
+	},
+	NewBootstrap: nil,
+	NewMetadata: func(m *stack.Machine) interface{} {
+		if m == nil {
+			return &Metadata{}
+		}
+
+		meta := &Metadata{}
+
+		return meta
+	},
+}
+
+var (
+	_ stack.Validator = (*Credential)(nil)
+	_ stack.Validator = (*Metadata)(nil)
+)
+
+// Credential represents credential information
+// that are required to deploy a Marathon app.
+type Credential struct {
+	URL                string `json:"url" bson:"url" hcl:"url"`
+	BasicAuthUser      string `json:"basic_auth_user" bson:"basic_auth_user" hcl:"basic_auth_user"`
+	BasicAuthPassowrd  string `json:"basic_auth_password" bson:"basic_auth_password" hcl:"basic_auth_password"`
+	RequestTimeout     int    `json:"request_timeout" bson:"request_timeout" hcl:"request_timeout"`
+	DeploymentTimepout int    `json:"deployment_timeout" bson:"deployment_timeout" hcl:"deployment_timeout"`
```

Comment:
> Anyhow for the sake of consistency with other providers, we can leave it as is.

Yes, the `meta`  object differ in notation starting from solo times. `jMachine.meta` always had fields that come from AWS (`storage_size`, `instance_type` etc.), the jCredentialDatas.meta always has the fields with underscores (`access_key`, `secret_key` etc.). So adding new providers I just followed what we had. Also it would be possible in future to get rid of 3 tags each struct has currently (`json`, `bson` and `hcl`) without additional mapping (snake case -> camel case).